src/main.ts

```diff
@@ -1,121 +1,135 @@
 
 export default class VimReadingViewNavigation extends Plugin {
-	settings: VimScrollSetting;
-	navScope: Scope;
-
-	async onload() {
-		await this.loadSettings();
-		this.addSettingTab(new VimScrollSettingTab(this.app, this));
-
-		this.navScope = new Scope(app.scope);
-		this.navScope.register([], 'j', (evt: KeyboardEvent) => {
-			const leaf = app.workspace.getActiveViewOfType(MarkdownView);
-			if (
-				leaf.getMode() === 'preview' &&
-				this.displayValue(leaf) === 'none'
-			) {
-				this.scrollDown(leaf);
-				return false;
-			}
-			return true;
-		});
-		this.navScope.register([], 'k', (evt: KeyboardEvent) => {
-			const leaf = app.workspace.getActiveViewOfType(MarkdownView);
-			if (
-				leaf.getMode() === 'preview' &&
-				this.displayValue(leaf) === 'none'
-			) {
-				this.scrollUp(leaf);
-				return false;
-			}
-			return true;
-		});
+    settings: VimScrollSetting;
+    navScope: Scope;
+
+    async onload() {
+        await this.loadSettings();
+        this.addSettingTab(new VimScrollSettingTab(this.app, this));
+        this.navScope = new Scope(app.scope);
+
+        // Scroll down
+        this.navScope.register([], 'j', (evt: KeyboardEvent) => {
+            const leaf = app.workspace.getActiveViewOfType(MarkdownView);
+            if (
+                leaf.getMode() === 'preview' &&
+                this.displayValue(leaf) === 'none'
+            ) {
+                this.scrollDown(leaf);
+                return false;
+            }
+            return true;
+        });
+
+        // Scroll up
+        this.navScope.register([], 'k', (evt: KeyboardEvent) => {
+            const leaf = app.workspace.getActiveViewOfType(MarkdownView);
+            if (
+                leaf.getMode() === 'preview' &&
+                this.displayValue(leaf) === 'none'
+            ) {
+                this.scrollUp(leaf);
+                return false;
+            }
+            return true;
+        });
+
+        // Jump to top
+        let keyArray: string[] = [];
+        const jumpTopEvent = (event: KeyboardEvent) => {
+            const leaf = app.workspace.getActiveViewOfType(MarkdownView);
+            if (
+                leaf.getMode() === 'preview' &&
+                this.displayValue(leaf) === 'none'
+            ) {
+                event.preventDefault();
+                if (event.key != 'g') {
+                    keyArray = [];
+                } else {
+                    keyArray.push(event.key);
+                    if (keyArray.length === 2) {
+                        leaf.previewMode.applyScroll(0);
+                        keyArray = [];
+                    }
+                }
+            }
+        }
+        addEventListener('keydown', jumpTopEvent, { capture: true });
+
+        // Jump to bottom
+        //  1st evt registers g when CapsLock is on
+        //  2nd evt registers Shift+g
         this.navScope.register([], 'g', (evt: KeyboardEvent) => {
```

Comment:
The first argument should be `[]`, not `{]`.

Reply:
Is it not `[]`?

This is what I see

![2022-08-10-193441_1920x1080_scrot](https://user-images.githubusercontent.com/47350090/183945911-8fad5920-de3d-4113-9815-739c305c5d8d.png)

Reply:
Ah sorry, it is. 

In some scroll positions, it didn't render it correctly on my screen. (I would take a screenshot, but as soon as I activate the tool, it renders correctly.)